### PR TITLE
Initialize Collection after storage permission granted

### LINKED ISSUE #8558: [Bug] Collection Not Opened [2.15]
I can reproduce it when I open the app after denying the storage permission. I think it was introduced in the April 5 commits, around 732eb1a.

This occurs due to the collection not being initialized before the```handleStartup()``` call once storage permission is granted. Hence, this leads to ```handleStartupFailure()``` being called & the dialog showing whenever the storage permission is granted. Collection is only initialized once getCol() is called. So this can be fixed by calling firstCollectionOpen() inside handleStartup(). We can remove the explicit firstCollectionOpen() call in DeckPicker's  onCreate(). Then, we can simply call handleStartup() once in onCreate(), and once when permission is granted.

Can I work on this?

_Originally posted by @farjadilyas in https://github.com/ankidroid/Anki-Android/issues/8554#issuecomment-817163082_

### PR DESCRIPTION
**Pull request by farjadilyas**
## Purpose / Description
Collection was not initialized after storage permission was newly granted

## Fixes
Fixes #8558 

## Approach
Made the following changes in DeckPicker
- firstCollectionOpen() is only responsible for checking if the collection can be opened. It isn't responsible for error checking anymore
- handleStartup() initializes collection using firstCollectionOpen immediately if permission granted
- handleStartup() requests for storage permission if it's not granted
- handleStartup() called in onCreate() & after permission granted

## How Has This Been Tested?
- Samsung Galaxy A80 API 30 (Physical Device)
- Pixel 3a XL API 27 (Emulator)

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

**Review by david-allison-1 — APPROVED** [review]
Tested - looks good to me, thanks!

**Comment by david-allison-1** [comment]
It'd be great to get a regression test for this path in another PR

**Comment by farjadilyas** [comment]
Ah, I didn't mean to merge master into this. Should I reset and push?

**Comment by david-allison-1** [comment]
In future: Could you use `rebase` instead of `merge` when fixing conflicts

**Comment by david-allison-1** [comment]
> Ah, I didn't mean to merge master into this while resolving a conflict. Should I reset and push?

Didn't see this, yeah. `--force-with-lease` is fine for PRs

**Comment by david-allison-1** [comment]
waiting for the conflict fix, both other than that, good to go

**Comment by farjadilyas** [comment]
done!

**Review by mikehardy — APPROVED** [review]
Nice! Thank you

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java [hunk 1]
```diff
@@ -449,15 +449,12 @@ protected void onCreate(Bundle savedInstanceState) throws SQLException {
...
+        
         mCustomStudyDialogFactory = new CustomStudyDialogFactory(this::getCol, this).attachToActivity(this);
 
-        SharedPreferences preferences = AnkiDroidApp.getSharedPrefs(getBaseContext());
-
         //we need to restore here, as we need it before super.onCreate() is called.
         restoreWelcomeMessage(savedInstanceState);
-        // Open Collection on UI thread while splash screen is showing
-        boolean colOpen = firstCollectionOpen();
+        handleStartup();
```
Comment (Akshay0701) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:457]: you should call this function at last of onCreate Function

Comment (Akshay0701) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:457]: anyway i will include this bug in my PR #8438 so don't worry about this one :)

Comment (farjadilyas) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:457]: Could you explain why?

Comment (Akshay0701) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:457]: i assuming its my device fault ,sorry :(

Comment (farjadilyas) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:457]: No worries!

**Pull request merged**